File: solutions/golang/loggingframework/log_level.go
package loggingframework

type LogLevel int

const (
	LogLevelDebug LogLevel = iota
	LogLevelInfo
	LogLevelWarning
	LogLevelError
	LogLevelFatal
)

func (l LogLevel) String() string {
	return [...]string{"DEBUG", "INFO", "WARNING", "ERROR", "FATAL"}[l]
}

File: solutions/typescript/src/LoggingFramework/LogLevel.ts
export enum LogLevelEnum {
  DEBUG, // 0
  INFO, // 1
  WARN, // 2
  ERROR, // 3
  FATAL, // 4
}

export function isAsSevereAs(
  level1: LogLevelEnum,
  level2: LogLevelEnum,
): boolean {
  return level1 >= level2;
}
